pkg/query/selection.go

```diff
@@ -109,13 +111,23 @@ type Criterion struct {
 	RightOp []string
 	// Type is the type of the query
 	Type CriterionType
+	// TemplateParametersMap contains the template parameters of Criterion supporting sub-queries
+	TemplateParametersMap map[string]interface{}
```

Comment:
I would create a type for that, e.g. `type TemplateParameters map[string]interface{}`, even if not I suggest removing the "Map" from the name (and the interface function).

Reply:
I've removed "Map" from the name.